content/200-concepts/100-components/01-prisma-schema/08-views.mdx

```diff
@@ -0,0 +1,180 @@
+
+</Admonition>
+
+Database views allow you to name and store a query. In relational databases, views are stored SQL queries that may include columns in multiple tables, or calculated values such as aggregates. In MongoDB, views are queryable objects where the contents are defined by an [aggregation pipeline](https://www.mongodb.com/docs/manual/core/aggregation-pipeline) on other collections.
+
+This guide explains how to enable the `views` preview feature and add views to your Prisma schema.
+
+</TopBlock>
+
+## How to enable the <inlineCode>views</inlineCode> preview feature
+
+Support for views is currently in an early preview. To enable the `views` preview feature, add the `views` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema file:
+
+```prisma file=schema.prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["views"]
+}
+```
+
+## How to add views to your Prisma schema
+
+To add a view to your Prisma schema, use the `view` keyword. For example, start with a `User` model with a linked `Profile`:
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma
+model User {
+  id      Int      @id @default(autoincrement())
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     Int    @id @default(autoincrement())
+  bio    String
+  user   User   @relation(fields: [userId], references: [id])
+  userId Int    @unique
+}
+```
+
+</tab>
+<tab>
+
+```prisma
+model User {
+  id      String   @id @default(auto()) @map("_id") @db.ObjectId
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  bio    String
+  userId String @unique @db.ObjectId
+  User   User   @relation(fields: [userId], references: [id])
+}
+```
+
+</tab>
+</TabbedContent>
+
+Next, take a `UserInfo` view in the underlying database that combines the `email` and `name` fields from the `User` model and the `bio` field from the `Profile` model.
+
+For a relational database, the SQL statement to create this view is:
+
+```sql
+CREATE VIEW "UserInfo" AS
+    SELECT u.id, email, name, bio
+    FROM "User" u
+    LEFT JOIN "Profile" p ON u.id = p."userId"
+```
+
+For MongoDB, you can [create a view](https://www.mongodb.com/docs/manual/core/views/join-collections-with-view/) with the following command:
+
+```t
+db.createView( "UserInfo", "User", [
+   {
+      $lookup:
+         {
+            from: "Profile",
+            localField: "_id",
+            foreignField: "userId",
+            as: "ProfileData"
+         }
+   },
+   {
+      $project:
+         {
+           _id: 1,
+           email: 1,
+           name: 1,
+           bio: "$ProfileData.bio"
+         }
+   },
+      { $unwind: "$bio" }
+] )
+```
```

Comment:
tabs to switch between the two again maybe?

Reply:
I considered it but they really need different intros